README.md
# Fandom-NFT-Marketplace-Frontend

Fandom is a multi-chain platform for fans to connect with their favorite celebrities and collect exclusive digital memorabilia.  This is a multi-chain platform for creators, collectors, and enthusiasts to buy, sell, and discover unique and valuable digital collectibles known as Non-Fungible Tokens (NFTs).
Fans can discover, collect, and engage with exclusive NFTs from their most cherished celebrities.

## Features
The NFT Marketplace provides a range of features to enhance your experience :-

- Browse and discover a diverse collection of NFTs from various categories.
- Wallet integration to securely store and manage your digital assets.
- Buy NFTs directly from artists and collectors.
- Sell your own NFTs to interested buyers.
- Detailed NFT profiles with descriptions, images, and transaction history.

  #### Mint NFT

  - Connect your Metamask wallet
  - Navigate to "Mint" section
  - Upload the image
  - Fill NFt attribute details
  - Click on "Mint NFT" button

  #### Sell NFT

  - Connect your Metamask wallet
  - Navigate to "Sell" section
  - You can see the list of NFT you have
  - Click on "Sell" button
  - Put the price of the NFT
  - Click on "Confirm"

  #### Buy NFT

  - Connect your Metamask wallet
  - Navigate to "Buy" section
  - You can see the list of NFT available for sell
  - Click on "Buy" button

## System Diagram

![System Diagram](./frontend/docs/system-diagram.png)

## Sequence diagram to mint NFT

![Mint Sequence Diagram](./frontend/docs/mint.png)

## Sequence diagram to sell NFT

![Sell Sequence Diagram](./frontend/docs/sell.png)

## Sequence diagram to buy NFT

![Buy Sequence Diagram](./frontend/docs/buy.png)

## Smart Contract Details
S. No. | Smart Contract | Chain | Contract Address |
--- | --- | --- | --- |
1 | Fandom NFT | Mumbai Matic | <a href="https://mumbai.polygonscan.com/address/0x7AA0768e2f046a351b24AC00CE43CA8d2BCbc968" target="_blank">https://mumbai.polygonscan.com/address/0x7AA0768e2f046a351b24AC00CE43CA8d2BCbc968</a> |
2 | NFT Marketplace | Mumbai Matic | <a href="https://mumbai.polygonscan.com/address/0xa1AB3cE05b0Ae95D8e5e7CB09Df315A3d4820d13" target="_blank">https://mumbai.polygonscan.com/address/0xa1AB3cE05b0Ae95D8e5e7CB09Df315A3d4820d13</a> |
3 | Fandom NFT | Sepolia | <a href="https://sepolia.etherscan.io/address/0x740b9A44545367F97f6840dCb8F0c882A79687B4" target="_blank">https://sepolia.etherscan.io/address/0x740b9A44545367F97f6840dCb8F0c882A79687B4</a> |
4 | NFT Marketplace | Sepolia | <a href="https://sepolia.etherscan.io/address/0xA5E49fc6555b0A7f9FAe5a1400D7e5Afd20E1dE3" target="_blank">https://sepolia.etherscan.io/address/0xA5E49fc6555b0A7f9FAe5a1400D7e5Afd20E1dE3</a> |
5 | Fandom NFT | Avalanche Fuji | <a href="https://testnet.snowtrace.io/address/0x740b9A44545367F97f6840dCb8F0c882A79687B4" target="_blank">https://testnet.snowtrace.io/address/0x740b9A44545367F97f6840dCb8F0c882A79687B4</a> |
6 | NFT Marketplace | Avalanche Fuji | <a href="https://testnet.snowtrace.io/address/0xA5E49fc6555b0A7f9FAe5a1400D7e5Afd20E1dE3" target="_blank">https://testnet.snowtrace.io/address/0xA5E49fc6555b0A7f9FAe5a1400D7e5Afd20E1dE3</a> |
7 | Fandom NFT | Hedera | <a href="https://hashscan.io/testnet/contract/0.0.1748907" target="_blank">https://hashscan.io/testnet/contract/0.0.1748907</a> |
8 | NFT Marketplace | Hedera | <a href="https://hashscan.io/testnet/contract/0.0.1749094" target="_blank">https://hashscan.io/testnet/contract/0.0.1749094</a> |

## Demo

<a href="https://fandom-nft-marketplace-v2.netlify.app/" target="_blank">https://fandom-nft-marketplace-v2.netlify.app/</a> 

## Setup

- Clone the repo:- `git clone https://gitlab.mindfire.co.in/bhupeshd/fandom-nft-marketplace-frontend.git`
- Create a .env file and take reference from .env.sample file to fill it
- Install NPM packages `npm install`
- Run to start your project  `npm start`
- Open this url on browser `http://localhost:3000/`


CHANGELOG.md

# Changelog

All notable changes to this project will be documented in this file. The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/), and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).

## [Unreleased]

### Added

- [List new features or enhancements added but not yet released]

### Changed

- [List changes, improvements, or modifications made but not yet released]

### Fixed

- [List bug fixes or corrections made but not yet released]

## [Version] - YYYY-MM-DD

### Added

- [List new features or enhancements added in this version]

### Changed

- [List changes, improvements, or modifications made in this version]

### Fixed

- [List bug fixes or corrections made in this version]

## [Older Versions]

[Repeat the above sections for older versions of your project]

## Contributing

Please read [CONTRIBUTING.md](./contributing-template.md) for details on our code of conduct, and the process for submitting pull requests to us.

## Authors

- B. Vinoth Raj
- Bhupesh Dubey

## License

This project is licensed under the [License Name] - see the [LICENSE.md](./license-template.md) file for details.

## Customization

You can customize this changelog to meet the specific needs and preferences of your project. Feel free to modify the structure, add more details, or change the formatting to better suit your project's versioning and documentation requirements.



src/utils/commonUtils.js
export const supportedChains = {
  '0x13881': 'Mumbai Matic',
  '0xaa36a7': 'Sepolia',
  '0xa869': 'Avalanche Fuji',
  '0x128': 'Hedera',
}

export const chainProperties = {
  '0x13881': {
    chain: 'Mumbai Matic',
    currency: 'MATIC',
    nftAddress: '0x7AA0768e2f046a351b24AC00CE43CA8d2BCbc968',
    marketplaceAddress: '0xa1AB3cE05b0Ae95D8e5e7CB09Df315A3d4820d13',
    explorerUrl: 'https://mumbai.polygonscan.com/tx/',
    decimal: 18,
  },
  '0xaa36a7': {
    chain: 'Sepolia',
    currency: 'ETH',
    nftAddress: '0x740b9A44545367F97f6840dCb8F0c882A79687B4',
    marketplaceAddress: '0xA5E49fc6555b0A7f9FAe5a1400D7e5Afd20E1dE3',
    explorerUrl: 'https://sepolia.etherscan.io/tx/',
    decimal: 18,
  },
  '0xa869': {
    chain: 'Avalanche Fuji',
    currency: 'AVAX',
    nftAddress: '0x740b9A44545367F97f6840dCb8F0c882A79687B4',
    marketplaceAddress: '0xA5E49fc6555b0A7f9FAe5a1400D7e5Afd20E1dE3',
    explorerUrl: 'https://testnet.snowtrace.io/tx/',
    decimal: 18,
  },
  '0x128': {
    chain: 'Hedera',
    currency: 'HBAR',
    nftAddress: '0x837e3146Ef1172254fa985ADCE891d50459A41B9',
    marketplaceAddress: '0x78fB6373a7fFB573dC50392d9eCa4b46bF3F7Ef7',
    explorerUrl: 'https://hashscan.io/testnet/transactions',
    decimal: 8,
  },
}

export const PINATA_GATEWAY_BASE_URL =
  'https://orange-scrawny-amphibian-531.mypinata.cloud/ipfs/'

export const PINATA_JSON_UPLOAD_URL =
  'https://api.pinata.cloud/pinning/pinJSONToIPFS'

export const PINATA_FILE_UPLOAD_URL =
  'https://api.pinata.cloud/pinning/pinFileToIPFS'

export const ALLOWED_IMAGE_FORMATS = ['jpg', 'jpeg', 'png', 'gif']

export const REGEX_FOR_PRICE = /^(?:\d{0,5}\.\d{1,5})$|^\d{0,5}$/
